File: src/main/java/api/ApiNode.java
package api;

public interface ApiNode {
    /**
     * 应返回 Api 节点路径，
     * 例如 http://example.com/hydev/jetty
     * 应返回 /hydev/jetty
     *
     * @return Api 节点路径
     */
    String path();

    /**
     * 处理 Api 调用
     *
     * @param access 客户端请求信息
     * @return 返回给客户端的内容
     */
    String process(ApiAccess access);
}

File: src/main/kotlin/org/hydev/ApiNode.kt
package org.hydev

/**
 * Each api node represents a specific location on the server.
 * (Eg. "http://.../hello/world" has the path "/hello/world")
 *
 * @author HyDEV Team (https://github.com/HyDevelop)
 * @author Hykilpikonna (https://github.com/hykilpikonna)
 * @author Vanilla (https://github.com/VergeDX)
 * @since 2020-07-03 12:34
 */
abstract class ApiNode(path: String, val isSecret: Boolean = false)
{
    val path = path.toLowerCase()

    abstract fun process(access: ApiAccess): Any?
}
